include/deal.II/lac/constraint_matrix.h

```diff
@@ -1406,6 +1421,8 @@ ConstraintMatrix::add_entry (const size_type line,
   // exists, since we don't want to enter it twice
   Assert (lines_cache[calculate_line_index(line)] != numbers::invalid_size_type,
           ExcInternalError());
+  Assert (!local_lines.size() || local_lines.is_element(column),
```

Comment:
You can always use a IndexSet other than the one obtained from locally relevant DoFs. In my opinion a constraint x_i=x_j is quite dangerous if j is not locally relevant. 
I understand your point, but is it really a problem to force the inclusion of such a non locally relevant DoF in the IndexSet of the ConstraintMatrix if this is indeed desired?